### PR TITLE
fix: cant delete connections due to fk constrain

### PR DESCRIPTION
**Pull request by ig-ra**
event is dependent on connection. Since event is soft-deleted, then connection should be soft-deleted as well

### CODE REVIEW

File: internal/backend/db/dbgorm/scheme/scheme.go [hunk 1]
```diff
@@ -76,6 +76,8 @@ type Connection struct {
 	StatusCode    int32 `gorm:"index"`
 	StatusMessage string
 
+	DeletedAt gorm.DeletedAt `gorm:"index"`
```
Comment (efiShtain) [inline internal/backend/db/dbgorm/scheme/scheme.go:79]: Don't you need to update queries accordingly to not fetch deleted?

Comment (ig-ra) [reply internal/backend/db/dbgorm/scheme/scheme.go:79]: gorm is handling this

Comment (ig-ra) [reply internal/backend/db/dbgorm/scheme/scheme.go:79]: there is this fixture - assertDeleted, which tests exactly this - tests that find query will skip deleted objects and raw db access will still find them

**Review by efiShtain — APPROVED** [review]
Approved

Please don't mix efforts in future PRs

**Pull request merged**